codegen/smithy-dafny-codegen/src/main/java/software/amazon/polymorph/smithyjava/nameresolver/Dafny.java

```diff
@@ -259,6 +339,19 @@ public CodeBlock typeDescriptor(ShapeId shapeId) {
                 return CodeBlock.of("$L", typeDescriptor);
             }
         }
+        if (shape.getType().isShapeType(ShapeType.LIST)) {
+            CodeBlock elementTypeDescriptor = typeDescriptor(shape.asListShape().get().getMember().getTarget());
+            return CodeBlock.of("$T._typeDescriptor($L)", Constants.DAFNY_SEQUENCE_CLASS_NAME, elementTypeDescriptor);
+        }
+        if (shape.getType().isShapeType(ShapeType.SET)) {
+            CodeBlock elementTypeDescriptor = typeDescriptor(shape.asSetShape().get().getMember().getTarget());
+            return CodeBlock.of("$T._typeDescriptor($L)", Constants.DAFNY_SET_CLASS_NAME, elementTypeDescriptor);
+        }
+        if (shape.getType().isShapeType(ShapeType.MAP)) {
+            CodeBlock keyTypeDescriptor = typeDescriptor(shape.asMapShape().get().getKey().getTarget());
+            CodeBlock valueTypeDescriptor = typeDescriptor(shape.asMapShape().get().getValue().getTarget());
+            return CodeBlock.of("$T._typeDescriptor($L, $L)", Constants.DAFNY_MAP_CLASS_NAME, keyTypeDescriptor, valueTypeDescriptor);
+        }
```

Comment:
Question: Where is the  Dafny 4.2.0 switch?

This is a public method.
So it can be called by another class that is generating for less than 4.2.0.

Is this logic invalid for Dafny less than 4.2.0?
If so, we need the check.

Reply:
This method just emits code to create type descriptor instances, and the way those are created didn't change, only where they are required parameters. So this logic is correct for any Dafny version.

Reply:
But we have separate logic for injecting Type Descriptors in to Aggregate Smithy Types, 
right?
Does this get called on those paths?
I guess so, but I bet that it was always called with the member,
not the aggregate, as the aggregate is either hard coded 
or was not needed...

Reply:
Yup, I just found there were cases of `Option<T>` and `Result<T, Error>` where `T` was driven by one of these aggregate shapes, so it was necessary to handle more cases in this method.